[V3] Migrating to AsyncLocalStorage. (#13)
odedglas/node-execution-context: README.md

## API

### run(fn: Function, context: unknown)

Runs given callback that will be exposed to the given context.

> The context will be exposed within all callbacks and promises chain that will be trigger from given fn.

Reviewer: ```suggestion
The context will be exposed to all callbacks and promise chains triggered from the given fn.
```

### get()

Gets the current asynchronous execution context.

> > This API may throw CONTEXT_DOES_NOT_EXISTS error if accessed without initializing the context properly.

Reviewer: change const to `CONTEXT_DOES_NOT_EXIST`

Author: you mean regarding the fact that the error message is not `CONTEXT_DOES_NOT_EXIST`?

### set(context: unknown)

Sets the current asynchronous execution context to given value.

> This API may throw CONTEXT_DOES_NOT_EXISTS error if accessed without initializing the context properly.

### create(context?: unknown)

Creates a given context for the current asynchronous execution.

> This method shouldn't be used most of the times and relevant only when you need to support some even driven case such and can't use the `run` method directly.

Reviewer: ```suggestion
It is recommended to use the `run` method. This method should be used in special cases in which the `run` method cannot be used directly.
```

| Code | When |
|-|-

> | CONTEXT_DOES_NOT_EXISTS | When try to `get` / `set` the context, but it yet been created.

Reviewer: ```suggestion
| CONTEXT_DOES_NOT_EXISTS | When attempting to `get` / `set` a context, that has not yet been created.
```
